clustermesh/clustermesh.go

```diff
@@ -156,6 +157,39 @@ func (k *K8sClusterMesh) apiserverImage() string {
 	return utils.BuildImagePath(k.params.ApiserverImage, defaults.ClusterMeshApiserverImage, k.params.ApiserverVersion, k.imageVersion)
 }
 
+func (k *K8sClusterMesh) etcdImage() string {
+	etcdVersion := "v3.4.13"
+	arch := runtime.GOARCH
+	if arch == "amd64" {
+		return "quay.io/coreos/etcd:" + etcdVersion
+	}
+	return "quay.io/coreos/etcd:" + etcdVersion + "-" + arch
+}
+
+func (k *K8sClusterMesh) etcdEnvs() []corev1.EnvVar {
+	envs := []corev1.EnvVar{
+		{
+			Name:  "ETCDCTL_API",
+			Value: "3",
+		},
+		{
+			Name: "HOSTNAME_IP",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "status.podIP",
+				},
+			},
+		},
+	}
+	if runtime.GOARCH == "arm64" {
```

Comment:
Same here, this should check the `GOARCH` for the cluster where Cilium is deployed.